{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3be61fb5-53ee-42ec-acc5-27168c369715",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import bioframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "41e64e53-865b-47ff-8c13-3fd08fe0f43f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read gtf file as bed9 and returns bed6\n",
    "\n",
    "def gtf_to_bed6(gtf):\n",
    "    gtf_1 = gtf[gtf['end'] == 'gene']\n",
    "    c1 = gtf_1['chrom']\n",
    "    c2 = gtf_1['name']\n",
    "    c3 = gtf_1['score']\n",
    "    c4 = gtf_1['rgb'].str.split(';',expand=True)[0].str.replace('gene_id ', '').str.replace('\"', '')\n",
    "    c5 = gtf_1['strand']\n",
    "    c6 = gtf_1['thickStart']\n",
    "    c7 = gtf_1['rgb'].str.split(';',expand=True)[2].str.replace('gene_name ', '').str.replace('\"', '').astype(str).str.replace(' ', '')\n",
    "    BED = pd.concat([c1,c2,c3,c4,c5,c6,c7],axis=1)\n",
    "    BED['strand'] = 1000\n",
    "    BED.columns = ['chrom','start','end','name','score','strand','gene_name']\n",
    "    BED_sorted = bioframe.sort_bedframe(BED)\n",
    "    return BED_sorted\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "11891173-bd26-4f60-ada8-ed5a3b4fb300",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/kimj7/miniconda3/envs/R_python_env/lib/python3.8/site-packages/bioframe/io/fileops.py:74: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(filepath_or, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "#mm10_gtf = bioframe.read_table('/mnt/g/genomics/mm10.refGene.gtf',schema='bed9')\n",
    "#mm10_known_gtf = bioframe.read_table('/mnt/g/genomics/mm10.knownGene.gtf',schema='bed9')\n",
    "#mm10_ncbi_gtf = bioframe.read_table('/mnt/g/genomics/mm10.ncbiRefSeq.gtf',schema='bed9')\n",
    "GRCm39_gtf = bioframe.read_table('/mnt/g/genomics/Mus_musculus.GRCm39.110_noHeader.gtf',schema='bed9')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c918907a-7314-4d29-9805-41455d2b1222",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gene_id \"ENSMUSG00000104478\"; gene_version \"2\"; gene_name \"Gm38212\"; gene_source \"havana\"; gene_biotype \"TEC\";'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GRCm39_gtf['rgb'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c11184e7-2e6e-4e46-bab3-75d75be89b67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3143476</td>\n",
       "      <td>3144545</td>\n",
       "      <td>ENSMUSG00000102693</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3172239</td>\n",
       "      <td>3172348</td>\n",
       "      <td>ENSMUSG00000064842</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>Gm26206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3276124</td>\n",
       "      <td>3741721</td>\n",
       "      <td>ENSMUSG00000051951</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>Xkr4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3322980</td>\n",
       "      <td>3323459</td>\n",
       "      <td>ENSMUSG00000102851</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>Gm18956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3435954</td>\n",
       "      <td>3438772</td>\n",
       "      <td>ENSMUSG00000103377</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>Gm37180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90676615</td>\n",
       "      <td>90678894</td>\n",
       "      <td>ENSMUSG00000099399</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>Gm28301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90763696</td>\n",
       "      <td>90766736</td>\n",
       "      <td>ENSMUSG00000095366</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>Gm21860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90764326</td>\n",
       "      <td>90774754</td>\n",
       "      <td>ENSMUSG00000095134</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>Mid1-ps1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56939</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90796007</td>\n",
       "      <td>90827734</td>\n",
       "      <td>ENSMUSG00000096768</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>Gm47283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56940</th>\n",
       "      <td>chrY</td>\n",
       "      <td>90848682</td>\n",
       "      <td>90855309</td>\n",
       "      <td>ENSMUSG00000099871</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>Gm21742</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>56941 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom     start       end                name  score strand  \\\n",
       "0      chr1   3143476   3144545  ENSMUSG00000102693   1000      +   \n",
       "1      chr1   3172239   3172348  ENSMUSG00000064842   1000      +   \n",
       "2      chr1   3276124   3741721  ENSMUSG00000051951   1000      -   \n",
       "3      chr1   3322980   3323459  ENSMUSG00000102851   1000      +   \n",
       "4      chr1   3435954   3438772  ENSMUSG00000103377   1000      -   \n",
       "...     ...       ...       ...                 ...    ...    ...   \n",
       "56936  chrY  90676615  90678894  ENSMUSG00000099399   1000      -   \n",
       "56937  chrY  90763696  90766736  ENSMUSG00000095366   1000      -   \n",
       "56938  chrY  90764326  90774754  ENSMUSG00000095134   1000      +   \n",
       "56939  chrY  90796007  90827734  ENSMUSG00000096768   1000      +   \n",
       "56940  chrY  90848682  90855309  ENSMUSG00000099871   1000      +   \n",
       "\n",
       "           gene_name  \n",
       "0      4933401J01Rik  \n",
       "1            Gm26206  \n",
       "2               Xkr4  \n",
       "3            Gm18956  \n",
       "4            Gm37180  \n",
       "...              ...  \n",
       "56936        Gm28301  \n",
       "56937        Gm21860  \n",
       "56938       Mid1-ps1  \n",
       "56939        Gm47283  \n",
       "56940        Gm21742  \n",
       "\n",
       "[56941 rows x 7 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert to bed6\n",
    "#mm10_bed6 = gtf_to_bed6(mm10_gtf)\n",
    "#mm10_known_bed6 = gtf_to_bed6(mm10_known_gtf)\n",
    "#mm10_ncbi_bed6 = gtf_to_bed6(mm10_ncbi_gtf)\n",
    "GRCm39_bed6 = gtf_to_bed6(GRCm39_gtf)\n",
    "GRCm39_bed6['chrom']= \"chr\"+(GRCm39_bed6['chrom']).astype(str)\n",
    "GRCm39_bed6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5cc65d30-5ada-41ee-bb6e-d0fedd09f623",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save as bed file\n",
    "#mm10_bed6.to_csv('/mnt/g/genomics/mm10.refGene_transcripts_bed6.bed',sep='\\t',index=False,header=False)\n",
    "#mm10_known_bed6.to_csv('/mnt/g/genomics/mm10.knownGene_transcripts_bed6.bed',sep='\\t',index=False,header=False)\n",
    "#mm10_ncbi_bed6.to_csv('/mnt/g/genomics/mm10.ncbi_transcripts_bed6.bed',sep='\\t',index=False,header=False)\n",
    "GRCm39_bed6.to_csv('/mnt/g/genomics/GRCm39_gene_bed7.bed',sep='\\t',index=False,header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "d45f157e-c630-4d2c-a7d8-59e86af65180",
   "metadata": {},
   "outputs": [],
   "source": [
    "# try reading as bed\n",
    "mm10_bed = bioframe.read_table('/mnt/g/genomics/GRCm39_gene_bed7.bed',schema='bed9')\n",
    "mm10_knownGene = bioframe.read_table('/mnt/g/genomics/mm10.knownGene_transcripts_bed6.bed',schema='bed9')\n",
    "mm10_ncbi = bioframe.read_table('/mnt/g/genomics/mm10.ncbi_transcripts_bed6.bed',schema='bed9')\n",
    "mm10_refGene = bioframe.read_table('/mnt/g/genomics/mm10.refGene_transcripts_bed6.bed',schema='bed9')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "816a806b-b7b0-44ff-bea3-da7b57e3253a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enrichedRegion</th>\n",
       "      <th>depletedRegion</th>\n",
       "      <th>geneShortName</th>\n",
       "      <th>geneId</th>\n",
       "      <th>enrichedFpkm</th>\n",
       "      <th>depletedFpkm</th>\n",
       "      <th>enrichmentFold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Scn4b</td>\n",
       "      <td>ENSMUSG00000046480</td>\n",
       "      <td>10.5906</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Lct</td>\n",
       "      <td>ENSMUSG00000026354</td>\n",
       "      <td>21.5010</td>\n",
       "      <td>0.046872</td>\n",
       "      <td>458.714422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Olfm3</td>\n",
       "      <td>ENSMUSG00000027965</td>\n",
       "      <td>20.3472</td>\n",
       "      <td>0.305437</td>\n",
       "      <td>66.616684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Gm10800</td>\n",
       "      <td>ENSMUSG00000075014</td>\n",
       "      <td>30.6210</td>\n",
       "      <td>0.557849</td>\n",
       "      <td>54.891198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Gm16869</td>\n",
       "      <td>ENSMUSG00000087580</td>\n",
       "      <td>11.3983</td>\n",
       "      <td>0.222871</td>\n",
       "      <td>51.143038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Dpysl5</td>\n",
       "      <td>ENSMUSG00000029168</td>\n",
       "      <td>14.4166</td>\n",
       "      <td>4.700120</td>\n",
       "      <td>3.067283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Flrt3</td>\n",
       "      <td>ENSMUSG00000051379</td>\n",
       "      <td>20.4417</td>\n",
       "      <td>6.711300</td>\n",
       "      <td>3.045863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Sdc2</td>\n",
       "      <td>ENSMUSG00000022261</td>\n",
       "      <td>15.6131</td>\n",
       "      <td>5.130840</td>\n",
       "      <td>3.042991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Marcksl1</td>\n",
       "      <td>ENSMUSG00000047945</td>\n",
       "      <td>26.0631</td>\n",
       "      <td>8.656510</td>\n",
       "      <td>3.010809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Crym</td>\n",
       "      <td>ENSMUSG00000030905</td>\n",
       "      <td>214.0550</td>\n",
       "      <td>71.349800</td>\n",
       "      <td>3.000078</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>279 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    enrichedRegion depletedRegion geneShortName              geneId  \\\n",
       "0           dorsal        ventral         Scn4b  ENSMUSG00000046480   \n",
       "1           dorsal        ventral           Lct  ENSMUSG00000026354   \n",
       "2           dorsal        ventral         Olfm3  ENSMUSG00000027965   \n",
       "3           dorsal        ventral       Gm10800  ENSMUSG00000075014   \n",
       "4           dorsal        ventral       Gm16869  ENSMUSG00000087580   \n",
       "..             ...            ...           ...                 ...   \n",
       "274        ventral         dorsal        Dpysl5  ENSMUSG00000029168   \n",
       "275        ventral         dorsal         Flrt3  ENSMUSG00000051379   \n",
       "276        ventral         dorsal          Sdc2  ENSMUSG00000022261   \n",
       "277        ventral         dorsal      Marcksl1  ENSMUSG00000047945   \n",
       "278        ventral         dorsal          Crym  ENSMUSG00000030905   \n",
       "\n",
       "     enrichedFpkm  depletedFpkm  enrichmentFold  \n",
       "0         10.5906      0.000000             inf  \n",
       "1         21.5010      0.046872      458.714422  \n",
       "2         20.3472      0.305437       66.616684  \n",
       "3         30.6210      0.557849       54.891198  \n",
       "4         11.3983      0.222871       51.143038  \n",
       "..            ...           ...             ...  \n",
       "274       14.4166      4.700120        3.067283  \n",
       "275       20.4417      6.711300        3.045863  \n",
       "276       15.6131      5.130840        3.042991  \n",
       "277       26.0631      8.656510        3.010809  \n",
       "278      214.0550     71.349800        3.000078  \n",
       "\n",
       "[279 rows x 7 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset\n",
    "gradient = pd.read_csv('/mnt/g/genomics/1-s2.0-S0896627315010867-mmc2.csv')\n",
    "gradient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "b1195d06-0093-4608-ab95-abbf247d63c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enrichedRegion</th>\n",
       "      <th>depletedRegion</th>\n",
       "      <th>geneShortName</th>\n",
       "      <th>geneId</th>\n",
       "      <th>enrichedFpkm</th>\n",
       "      <th>depletedFpkm</th>\n",
       "      <th>enrichmentFold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Gm10800</td>\n",
       "      <td>ENSMUSG00000075014</td>\n",
       "      <td>30.6210</td>\n",
       "      <td>0.557849</td>\n",
       "      <td>54.891198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Gm16869</td>\n",
       "      <td>ENSMUSG00000087580</td>\n",
       "      <td>11.3983</td>\n",
       "      <td>0.222871</td>\n",
       "      <td>51.143038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>4930429B21Rik</td>\n",
       "      <td>ENSMUSG00000049507</td>\n",
       "      <td>23.9162</td>\n",
       "      <td>4.426820</td>\n",
       "      <td>5.402569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>2310002F09Rik</td>\n",
       "      <td>ENSMUSG00000085370</td>\n",
       "      <td>16.7656</td>\n",
       "      <td>4.704820</td>\n",
       "      <td>3.563494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Gm10664</td>\n",
       "      <td>ENSMUSG00000074303</td>\n",
       "      <td>10.9264</td>\n",
       "      <td>0.035524</td>\n",
       "      <td>307.579707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Eid1</td>\n",
       "      <td>ENSMUSG00000078135</td>\n",
       "      <td>40.6391</td>\n",
       "      <td>7.729790</td>\n",
       "      <td>5.257465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Ptprg</td>\n",
       "      <td>ENSMUSG00000021745</td>\n",
       "      <td>166.9430</td>\n",
       "      <td>54.301200</td>\n",
       "      <td>3.074389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    enrichedRegion depletedRegion  geneShortName              geneId  \\\n",
       "3           dorsal        ventral        Gm10800  ENSMUSG00000075014   \n",
       "4           dorsal        ventral        Gm16869  ENSMUSG00000087580   \n",
       "29          dorsal        ventral  4930429B21Rik  ENSMUSG00000049507   \n",
       "70          dorsal        ventral  2310002F09Rik  ENSMUSG00000085370   \n",
       "111        ventral         dorsal        Gm10664  ENSMUSG00000074303   \n",
       "194        ventral         dorsal           Eid1  ENSMUSG00000078135   \n",
       "273        ventral         dorsal          Ptprg  ENSMUSG00000021745   \n",
       "\n",
       "     enrichedFpkm  depletedFpkm  enrichmentFold  \n",
       "3         30.6210      0.557849       54.891198  \n",
       "4         11.3983      0.222871       51.143038  \n",
       "29        23.9162      4.426820        5.402569  \n",
       "70        16.7656      4.704820        3.563494  \n",
       "111       10.9264      0.035524      307.579707  \n",
       "194       40.6391      7.729790        5.257465  \n",
       "273      166.9430     54.301200        3.074389  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gradient[np.invert(gradient['geneId'].isin(list(mm10_bed['name'])))]\n",
    "#gradient[np.invert(gradient['geneId'].isin(list(ens_1)))]\n",
    "#gradient[np.invert(gradient['geneId'].isin(['ENSMUSG00000046480']))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "b50ba9eb-9c3e-4350-a413-ca7e39991bd1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enrichedRegion</th>\n",
       "      <th>depletedRegion</th>\n",
       "      <th>geneShortName</th>\n",
       "      <th>geneId</th>\n",
       "      <th>enrichedFpkm</th>\n",
       "      <th>depletedFpkm</th>\n",
       "      <th>enrichmentFold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Gm10800</td>\n",
       "      <td>ENSMUSG00000075014</td>\n",
       "      <td>30.6210</td>\n",
       "      <td>0.557849</td>\n",
       "      <td>54.891198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>Gm16869</td>\n",
       "      <td>ENSMUSG00000087580</td>\n",
       "      <td>11.3983</td>\n",
       "      <td>0.222871</td>\n",
       "      <td>51.143038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>dorsal</td>\n",
       "      <td>ventral</td>\n",
       "      <td>4930429B21Rik</td>\n",
       "      <td>ENSMUSG00000049507</td>\n",
       "      <td>23.9162</td>\n",
       "      <td>4.426820</td>\n",
       "      <td>5.402569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Gm10664</td>\n",
       "      <td>ENSMUSG00000074303</td>\n",
       "      <td>10.9264</td>\n",
       "      <td>0.035524</td>\n",
       "      <td>307.579707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>ventral</td>\n",
       "      <td>dorsal</td>\n",
       "      <td>Ptprg</td>\n",
       "      <td>ENSMUSG00000021745</td>\n",
       "      <td>166.9430</td>\n",
       "      <td>54.301200</td>\n",
       "      <td>3.074389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    enrichedRegion depletedRegion  geneShortName              geneId  \\\n",
       "3           dorsal        ventral        Gm10800  ENSMUSG00000075014   \n",
       "4           dorsal        ventral        Gm16869  ENSMUSG00000087580   \n",
       "29          dorsal        ventral  4930429B21Rik  ENSMUSG00000049507   \n",
       "111        ventral         dorsal        Gm10664  ENSMUSG00000074303   \n",
       "273        ventral         dorsal          Ptprg  ENSMUSG00000021745   \n",
       "\n",
       "     enrichedFpkm  depletedFpkm  enrichmentFold  \n",
       "3         30.6210      0.557849       54.891198  \n",
       "4         11.3983      0.222871       51.143038  \n",
       "29        23.9162      4.426820        5.402569  \n",
       "111       10.9264      0.035524      307.579707  \n",
       "273      166.9430     54.301200        3.074389  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### genes not in the ENS data \n",
    "tf1 = gradient['geneShortName'].isin(list(mm10_bed['thickStart']))\n",
    "tf2 = gradient['geneId'].isin(list(mm10_bed['name']))\n",
    "\n",
    "absentee = gradient[np.invert(np.logical_or(tf1,tf2))]\n",
    "#gradient[np.invert(gradient['geneShortName'].isin(list(mm10_bed['thickStart'])))]\n",
    "absentee\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "973599f7-497c-4ce3-a8f1-190f79df391b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4933401J01Rik'"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
